scipy/special/specfun/specfun.f

```diff
@@ -6191,28 +6191,35 @@ SUBROUTINE HYGFZ(A,B,C,Z,ZHF,ISFER)
 C
         IMPLICIT DOUBLE PRECISION (A-H,O-Y)
         IMPLICIT COMPLEX *16 (Z)
-        LOGICAL L0,L1,L2,L3,L4,L5,L6
+        LOGICAL L0,L1,L2,L3,L4,L5,L6,L7
         X=DBLE(Z)
         Y=DIMAG(Z)
         EPS=1.0D-15
         ISFER=0
-        L0=C.EQ.INT(C).AND.C.LT.0.0D0
+        L0=C.EQ.INT(C).AND.C.LE.0.0D0
```

Comment:
Hm... this problem was here already, but checking `C.EQ.INT(C)` is bad practice because `INT(C)` overflows before `C` does. The `floor` check in the C wrapper is better.

Reply:
Would it be better to use a bigger sized integer in the fortran code? Something like IDINT? Or does the `floor` function itself have better overflow protections?

My only concern here is with duplicating code in both the wrapper file and the fortran file which seems like it would make the code base hard to follow and maintain.

Reply:
But to be clear I can also re-produce my checks in the c wrapper code if that's what needs to be done...

Reply:
> Or does the floor function itself have better overflow protections?

The idea is that in C `floor` returns a double, so overflow is not an issue. The closest Fortran functions `CEILING` and `FLOOR` return integers. I don't know of a way to round in Fortran that returns `REAL(8)`.

So I think this change is fine; the code was already doing the checks like that anyway.

Reply:
Just for clarification, you think *my* change is fine.  I do *not* need to revert the safety checks that I removed from specfun_wrappers.c

Reply:
> Just for clarification, you think my change is fine. I do not need to revert the safety checks that I removed from specfun_wrappers.c

Yes.